zephyr/samples/coredumps/src/main.c

```diff
@@ -0,0 +1,80 @@
+#include <zephyr/kernel.h>
+#include <zephyr/logging/log.h>
+
+#include "wifi.h"
+#include "zephyr/debug/coredump.h"
+
+#include "zephyr/drivers/gpio.h"
+#include <zephyr/device.h>
+
+LOG_MODULE_REGISTER(MAIN, LOG_LEVEL_INF);
+
+#define SW0_NODE DT_ALIAS(sw0)
+#if !DT_NODE_HAS_STATUS_OKAY(SW0_NODE)
+#error "Unsupported board: sw2 devicetree alias is not defined"
+#endif
+
+static const struct gpio_dt_spec button = GPIO_DT_SPEC_GET_OR(SW0_NODE, gpios, { 0 });
+static struct gpio_callback button_cb_data;
+static int prepare_button();
+void check_coredump();
```

Comment:
```suggestion
static const struct gpio_dt_spec button = GPIO_DT_SPEC_GET_OR(SW0_NODE, gpios, { 0 });
static struct gpio_callback button_cb_data;

static int prepare_button();
```

`check_coredump` seems unused and I'd prefer to separate global variables from function declarations by a newline.